qiskit/transpiler/layout.py

```diff
@@ -276,36 +268,35 @@ def from_intlist(int_list, *qregs):
         main_idx = 0
         for qreg in qregs:
             for idx in range(qreg.size):
-                out[(qreg, idx)] = int_list[main_idx]
+                out[QuBit(qreg, idx)] = int_list[main_idx]
                 main_idx += 1
         if main_idx != len(int_list):
             for int_item in int_list[main_idx:]:
                 out[int_item] = None
         return out
 
     @staticmethod
-    def from_tuplelist(tuple_list):
+    def from_qubit_list(qubit_list):
```

Comment:
Although (or perhaps because) I'm still unfamiliar with the context, I like this name change :)